crates/subspace-farmer/src/mock_rpc.rs

```diff
@@ -0,0 +1,86 @@
+use crate::rpc::{Error as MockError, NewHead, RpcClient};
+use async_trait::async_trait;
+use log::info;
+use subspace_core_primitives::Tag;
+use subspace_rpc_primitives::{
+    EncodedBlockWithObjectMapping, FarmerMetadata, SlotInfo, SolutionResponse,
+};
+use tokio::sync::mpsc;
+
+/// `MockRpc` wrapper.
+#[derive(Debug)]
+pub struct MockRpc {
+    metadata_recv: mpsc::Receiver<FarmerMetadata>,
+    block_recv: mpsc::Receiver<EncodedBlockWithObjectMapping>,
+    newhead_recv: mpsc::Receiver<NewHead>,
+    slot_recv: mpsc::Receiver<SlotInfo>,
+    tag_recv: mpsc::Receiver<Tag>,
+}
+
+impl MockRpc {
+    /// Create a new instance of [`MockRPC`].
+    pub(crate) fn new(
+        metadata_recv: mpsc::Receiver<FarmerMetadata>,
+        block_recv: mpsc::Receiver<EncodedBlockWithObjectMapping>,
+        newhead_recv: mpsc::Receiver<NewHead>,
+        slot_recv: mpsc::Receiver<SlotInfo>,
+        tag_recv: mpsc::Receiver<Tag>,
+    ) -> Self {
+        MockRpc {
```

Comment:
```suggestion
        Self {
```

Reply:
I don't have a side on this. But here is my thought on it:
Using the word `Self` makes me think of an *existing* object. Since this is a constructor, and should create a *new* object from scratch, using the full name instead of `Self` made more sense to me, and I'm inclined to use this practice. However, please correct/enlighten me if I'm wrong :)

Reply:
Actually, there is no golden rule about this point, it has both cons and pros, see the related discussion https://github.com/rust-unofficial/patterns/pull/61 . Just a suggestion, you can leave it if you prefer another way. From what I see, it seemingly has become sort of idiomatic these days.

Reply:
Using it when it is not line-wise far from the `impl` declaration, makes sense for the refactoring, and also does not hurt readability that much. Thanks for the link!